src/BoundaryConditions/fill_halo_regions.jl

```diff
@@ -27,18 +27,32 @@ for dir in (:west, :east, :south, :north, :bottom, :top)
         @inline $extract_bc(bc::Tuple) = $extract_bc.(bc)
     end
 end
-
+# For inhomogeneous BC we extract the _last_ one 
+# example 
+# `bc.west <: DCBC`
+# `bc.east <: PBC`
+# `extract_west_or_east_bc(bc) == bc.west`
+    
+  extract_west_or_east_bc(bc) = sort([bc.west,   bc.east],  lt=fill_first)[2]
+extract_south_or_north_bc(bc) = sort([bc.south,  bc.north], lt=fill_first)[2]
+ extract_bottom_or_top_bc(bc) = sort([bc.bottom, bc.top],   lt=fill_first)[2]
```

Comment:
Does this allocate? Using a tuple instead of array might not, eg `(bc.bottom, bc.top)`

Reply:
I' ve changed it using `max`

Reply:
good on ya